packages/world/src/modules/core/implementations/AccessManagementSystem.sol

```diff
@@ -3,22 +3,30 @@ pragma solidity >=0.8.21;
 
 import { System } from "../../../System.sol";
 import { AccessControl } from "../../../AccessControl.sol";
-import { ResourceId } from "../../../WorldResourceId.sol";
+import { ResourceId, ResourceIdInstance, WorldResourceIdInstance } from "../../../WorldResourceId.sol";
+import { RESOURCE_NAMESPACE } from "../../../worldResourceTypes.sol";
+import { IWorldErrors } from "../../../IWorldErrors.sol";
 import { ResourceAccess } from "../../../codegen/tables/ResourceAccess.sol";
 import { NamespaceOwner } from "../../../codegen/tables/NamespaceOwner.sol";
 
 /**
  * @title Access Management System
  * @dev This contract manages the granting and revoking of access from/to resources.
  */
-contract AccessManagementSystem is System {
+contract AccessManagementSystem is System, IWorldErrors {
+  using ResourceIdInstance for ResourceId;
+  using WorldResourceIdInstance for ResourceId;
+
   /**
    * @notice Grant access to the resource at the given resource ID.
    * @dev Requires the caller to own the namespace.
    * @param resourceId The ID of the resource to grant access to.
    * @param grantee The address to which access should be granted.
    */
   function grantAccess(ResourceId resourceId, address grantee) public virtual {
+    // Require the resource to exist
+    AccessControl.requireExistence(resourceId);
+
     // Require the caller to own the namespace
     AccessControl.requireOwner(resourceId, _msgSender());
```

Comment:
would this check satisfy `requireExistence`? a non-existent resource would be "owned" by address zero, this fail this `requireOwner` check?

I guess adding this check here means the error is much more accurate/actionable

Reply:
I had the same thought, but having the explicit check gives a better error message and is less fragile.

I guess the "theme" of this PR is methodically adding these checks everywhere, so you know the requirements without having to "figure it out" from context.